stackslib/src/burnchains/tests/mod.rs

```diff
@@ -468,15 +479,37 @@ impl TestBurnchainBlock {
         txop.txid =
             Txid::from_test_data(txop.block_height, txop.vtxindex, &txop.burn_header_hash, 0);
 
-        txop.memo = vec![STACKS_EPOCH_2_1_MARKER << 3];
+        txop.memo = vec![epoch_marker << 3];
         self.txs
             .push(BlockstackOperationType::LeaderBlockCommit(txop.clone()));
 
         miner.block_commits.push(txop.clone());
         txop
     }
 
-    // TODO: user burn support
+    /// Add an epoch 2.x block-commit
+    pub fn add_leader_block_commit(
+        &mut self,
+        ic: &SortitionDBConn,
+        miner: &mut TestMiner,
+        block_hash: &BlockHeaderHash,
+        burn_fee: u64,
+        leader_key: &LeaderKeyRegisterOp,
+        fork_snapshot: Option<&BlockSnapshot>,
+        parent_block_snapshot: Option<&BlockSnapshot>,
+    ) -> LeaderBlockCommitOp {
+        self.inner_add_block_commit(
```

Comment:
This is mostly out of curiosity than a complaint...why did you chose to separate this into a function directly wrapping another function?

Reply:
Because `TestBurnchainBlock::add_nakamoto_tenure_commit()` also calls this function.  The difference between the two is that `TestBurnchainBlock::add_nakamoto_tenure_commit()` converts the `StacksBlockId` into a `BlockHeaderHash`, which is what the `LeaderBlockCommitOp` ultimately expects.  The algorithm is otherwise the same for both functions; it's just that the argument types are different.